### PR TITLE
Extend test for unpack of multibitvector to cover all aspects variant…

### PR DESCRIPTION
**Pull request by baldersheim**
…s for OR.

@havardpe PR

### CODE REVIEW

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp [hunk 1]
```diff
@@ -253,6 +255,63 @@ Test::testThatOptimizePreservesUnpack()
     fixup_bitvectors();
 }
 
+void verifyOrUnpack(SearchIterator & s, const TermFieldMatchData * tfmd) {
+    s.initFullRange();
+    s.seek(1);
+    for (size_t i = 0; i < 3; i++) {
+        EXPECT_EQUAL(0u, tfmd[0].getDocId());
```
Comment (havardpe) [inline searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp:262]: consider using i

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp [hunk 2]
```diff
@@ -253,6 +255,63 @@ Test::testThatOptimizePreservesUnpack()
...
+    }
+    s.unpack(1);
+    EXPECT_EQUAL(0u, tfmd[0].getDocId());
+    EXPECT_EQUAL(1u, tfmd[1].getDocId());
+    EXPECT_EQUAL(0u, tfmd[2].getDocId());
+}
+
+void
+Test::testUnpackOfOr() {
+    _bvs[0]->clearBit(1);
```
Comment (havardpe) [inline searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp:272]: not fixed by fixup

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp [hunk 3]
```diff
@@ -253,6 +255,63 @@ Test::testThatOptimizePreservesUnpack()
...
+    _bvs[1]->setBit(1);
+    _bvs[2]->clearBit(1);
```
Comment (havardpe) [inline searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp:274]: not fixed by fixup

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp [hunk 4]
```diff
@@ -253,6 +255,63 @@ Test::testThatOptimizePreservesUnpack()
...
+        tfmd.resetOnlyDocId(0);
+    }
+
+    const MultiSearch * ms = dynamic_cast<const MultiSearch *>(s.get());
+    EXPECT_TRUE(ms != nullptr);
+    EXPECT_EQUAL(3u, ms->getChildren().size());
+
+    s = MultiBitVectorIteratorBase::optimize(std::move(s));
+    s->initFullRange();
+    ms = dynamic_cast<const MultiSearch *>(s.get());
```
Comment (havardpe) [inline searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp:308]: consider checking for multi-bitvector optimization

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp]: Fixed

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp:274]: Fixed

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp (hunk 3 above)
Comment (baldersheim) [reply searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp:276]: Fixed

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp (hunk 4 above)
Comment (baldersheim) [reply searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp]: Fixed

**Pull request merged**